### PR TITLE
add documentation for running knative on IBM s390x/IBM ppc64le platforms

### PR DESCRIPTION
**Pull request by dilipgb**
This document updates the installation steps needed to install knative on IBM s390x and IBM ppc64le platforms.

### CODE REVIEW

**Comment by davidhadas** [comment]
I find the doc confusing:
1. It is unclear how Kourier differ from "3-scale-kourier-gateway" 
2. It is unclear how to merge the instructions provided to the "3-scale-kourier-gateway" with those already shown for Kourier

I would suggest taking a straight forward approach and stating in the Note something like: "Use [this alternative documentation](https://deploy-preview-6043--knative.netlify.app/docs/install/yaml-install/serving/install-serving-with-yaml-on-IBM-Z-and-IBM-P/) to install a networking layer on IBM Z and IBM Power platforms."

Then in the documentation, I would suggest to include all steps needed to get to a working networking layer. 

Instead of "Customizing 3-scale-kourier-gateway on IBM Z and IBM P platform" as a header, I would use "Install a networking layer on IBM Z and IBM Power platforms."

Instead of "To run 3-scale-kourier-gateway on IBM Z and IBM P platforms, ..." I suggest "To install a networking layer on IBM Z and IBM P platforms, ..." and spell out the steps to do that

**Comment by dilipgb** [comment]
@davidhadas **3-scale-kourier-gateway** is the deployment name for kourier gateway deployed as part of the kourier deployment. I used the same deployment name, so users who face issues pulling images can relate to.

I will create a single document instead of redirecting user multiple times to different documentations. But there is no difference between standard deployment and deployment we are suggesting. We are patching the image for the same deployment that is done via standard knative scripts.

**Comment by dilipgb** [comment]
@davidhadas can you please review and let me know if this looks good? I updated the document to be more readable and why we need to patch the 3-scale-kourier-gateway.

File: docs/install/yaml-install/serving/install-serving-with-yaml-on-IBM-Z-and-IBM-P.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Install a networking layer on IBM Z and IBM Power platforms
+
+  Patch the deployment **deploy/3-scale-kourier gateway in kourier-system namespace** with latest available image from [redhat catalog](https://catalog.redhat.com/software/containers/openshift-service-mesh/proxyv2-rhel8/5d2cda455a134672890f640a). Check the latest tag and click on the get this image tag to copy the image name.
```
Comment (davidhadas) [inline docs/install/yaml-install/serving/install-serving-with-yaml-on-IBM-Z-and-IBM-P.md:3]:
```suggestion
This additional step is required for installing the Kourier networking layer on IBM Z and IBM Power platforms.

Once you completed installing Kourier,  patch the envoy image as described below. Use the envoy image included as part of the (RedHat Maistra)[https://maistra.io/] distribution. Maistra is an opinionated distribution of Istio designed to work with OpenShift and is supported on IBM Z and IBM Power platforms.

1. Find the image name to use:  
1a. Access the [redhat catalog](https://catalog.redhat.com/software/containers/openshift-service-mesh/proxyv2-rhel8/5d2cda455a134672890f640a) 
1b. Choose the appropriate architecture (ppc64le/s390x) from the architecture drop down. 
1c. Choose the latest tag from the list of available tags at the tag dropdown
1d. Click on the "get this image" tab and copy the image name. For example: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8@sha256:ced904...`

2. Patch the **3scale-kourier-gateway**  deployment in **kourier-system** namespace with latest available image using **kubectl** as shown below.
```

File: docs/install/yaml-install/serving/install-serving-with-yaml-on-IBM-Z-and-IBM-P.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Install a networking layer on IBM Z and IBM Power platforms
+
+  Patch the deployment **deploy/3-scale-kourier gateway in kourier-system namespace** with latest available image from [redhat catalog](https://catalog.redhat.com/software/containers/openshift-service-mesh/proxyv2-rhel8/5d2cda455a134672890f640a). Check the latest tag and click on the get this image tag to copy the image name. 
+
+  ```bash
+    kubectl patch deployment 3scale-kourier-gateway -n kourier-system -p \
+    '{"spec":{"template":{"spec":{"containers":[{"name":"kourier-gateway" "image":"<replace the proxyv2 image here>"}]}}}}'
```
Comment (davidhadas) [inline docs/install/yaml-install/serving/install-serving-with-yaml-on-IBM-Z-and-IBM-P.md:7]:
```suggestion
    '{"spec":{"template":{"spec":{"containers":[{"name":"kourier-gateway" "image":"<the proxyv2 image name>"}]}}}}'
```

File: docs/install/yaml-install/serving/install-serving-with-yaml.md [hunk 3]
```diff
@@ -29,6 +29,10 @@ To install the Knative Serving component:
 The following tabs expand to show instructions for installing a networking layer.
 Follow the procedure for the networking layer of your choice:
 
+!!! note
+    **Only Kourier network plugin supported for IBM Z and IBM Power platform.** 
+    Follow the below steps to install kourier. Post installation, [Use this alternative documentation](./install-serving-with-yaml-on-IBM-Z-and-IBM-P.md) to patch envoy image since we do not have envoy image support on IBM Z and IBM Power platforms, we will patch the envoy image based on RedHat Maistra.
```
Comment (davidhadas) [inline docs/install/yaml-install/serving/install-serving-with-yaml.md:34]:
```suggestion
    Follow the below steps to install Kourier. Post installation, patch the envoy image based on RedHat Maistra as described in [this link](./install-serving-with-yaml-on-IBM-Z-and-IBM-P.md).
```

**Review by davidhadas — CHANGES_REQUESTED** [review]
Please consider the below suggested changes.
Please verify that the provided instructions and overall guidance are correct and accurate.

**Comment by dilipgb** [comment]
@davidhadas I think sub point in docs/install/yaml-install/serving/install-serving-with-yaml-on-IBM-Z-and-IBM-P.md is not rendering properly in the deploy preview. I will see how to add them in markup and commit it.

**Comment by dilipgb** [comment]
@davidhadas I have corrected the alignment. Can you please review it again?

File: docs/install/yaml-install/serving/install-serving-with-yaml-on-IBM-Z-and-IBM-P.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Install a networking layer on IBM Z and IBM Power platforms
+
+This additional step is required for installing the Kourier networking layer on IBM Z and IBM Power platforms.
+
+Once you completed installing Kourier,  patch the envoy image as described below. Use the envoy image included as part of the (RedHat Maistra)[https://maistra.io/] distribution. Maistra is an opinionated distribution of Istio designed to work with OpenShift and is supported on IBM Z and IBM Power platforms.
```
Comment (davidhadas) [inline docs/install/yaml-install/serving/install-serving-with-yaml-on-IBM-Z-and-IBM-P.md:5]:
```suggestion
Once you completed installing Kourier,  patch the envoy image as described below. Use the envoy image included as part of the [RedHat Maistra](https://maistra.io/) distribution. Maistra is an opinionated distribution of Istio designed to work with OpenShift and is supported on IBM Z and IBM Power platforms.
```

**Comment by davidhadas** [comment]
/lgtm
@skonto, @Cali0707 can you pls have a look?

**Review by davidhadas — APPROVED** [review]

**Pull request merged**